### PR TITLE
feat(CORSO-49): view a specific services details

### PR DESCRIPTION
**Pull request by KarinaSofia**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CORSO-49?atlOrigin=eyJpIjoiMTcxZTA2MWQwMTIwNDRlMTllMTg4ODg1ZTMyMTc0MzIiLCJwIjoiaiJ9

## Context:
View a specfic service's details

## Changes
- implemented the front end for the view service details
- implemented error handling for an invalid id
- implemented testing, end to end, and for the error handling
- implemented the footer for the main page

## Before and After UI (Required for UI-impacting PRs)
no page before
#After
View service details
image

Footer
image

### CODE REVIEW

File: corso_website_fe/src/Main Page Components/BannerMain.css [hunk 1]
```diff
@@ -71,4 +71,9 @@
     line-height: normal;
     text-transform: uppercase;
     background: none;
-}
\ No newline at end of file
+}
+
+.row-banner{
```
Comment (DylanBrass) [inline corso_website_fe/src/Main Page Components/BannerMain.css:76]: What does this do ?

File: corso_website_fe/src/Pages/ViewServiceDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+function ViewServiceDetails() {
+    const { serviceId } = useParams<{ serviceId: string }>();
+    const [serviceDetails, setServiceDetails] = useState<ServiceDetails | null>(null);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        const fetchServiceDetails = async () => {
+            try {
+                const response = await axios.get(`http://localhost:8080/api/v1/corso/services/${serviceId}`);
```
Comment (DylanBrass) [inline corso_website_fe/src/Pages/ViewServiceDetails.tsx:22]: Probably should remove reminders lol

File: corso_website_fe/src/Main Page Components/AllServices.tsx [hunk 3]
```diff
@@ -41,7 +41,9 @@ function AllServices() {
                             </div>
                             <div className="service-info">
                                 <h2 className="service-name">{service.serviceName}</h2>
-                                {/* <a className="details-hyperlink" href="#">View Details</a> */}
```
Comment (DylanBrass) [inline corso_website_fe/src/Main Page Components/AllServices.tsx:44]: Is this meant to be commented?

File: corso_website_fe/src/Pages/ServiceDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import NavigationBar from "../Components/NavigationBar";
+import ViewServiceDetails from "./ViewServiceDetails";
+
+function ServiceDetailsPage() {
+
+    return (
+        <div>
+            <NavigationBar />
+            <ViewServiceDetails />
+        </div>
```
Comment (DylanBrass) [inline corso_website_fe/src/Pages/ServiceDetailsPage.tsx:10]: Should we put the footer here ?

File: corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/ViewServiceDetailsTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static com.codeborne.selenide.Selenide.*;
+import static com.codeborne.selenide.Condition.visible;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+public class ViewServiceDetailsTest {
```
Comment (DylanBrass) [inline corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/ViewServiceDetailsTest.java:16]: Did you exclude this test for the ci/cd

**Review by DylanBrass — APPROVED** [review]
Some things to fix at your discretion.

File: corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/ViewServiceDetailsTest.java (hunk 5 above)
Comment (KarinaSofia) [reply corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/ViewServiceDetailsTest.java:16]: nope i forgot by accident, just pushed it!

File: corso_website_fe/src/Pages/ServiceDetailsPage.tsx (hunk 4 above)
Comment (KarinaSofia) [reply corso_website_fe/src/Pages/ServiceDetailsPage.tsx:10]: i excluded it here to match the figma

File: corso_website_fe/src/Main Page Components/BannerMain.css (hunk 1 above)
Comment (KarinaSofia) [reply corso_website_fe/src/Main Page Components/BannerMain.css:76]: stops a sliding bar from appearing at the bottom

File: corso_website_fe/src/Main Page Components/AllServices.tsx (hunk 3 above)
Comment (KarinaSofia) [reply corso_website_fe/src/Main Page Components/AllServices.tsx:44]: i had to comment this out in my previous pr as it was blocking the check. for this pr I added the proper implementation, therefore I removed my comment. so to answer your question, yes it was meant to be commented out

**Review by Mila5847 — APPROVED** [review]
Service and footer looks great, testing is also done, some comment to be removed but they were mentioned.

**Pull request merged**